### ISSUE TITLE
Solarflare driver segfaults on lib.nfv.config selftest

### ISSUE THREAD

**Issue by eugeneia** [SnabbCo/snabbswitch#643]
Solarflare driver segfaults on  `lib.nfv.config` selftest. To reproduce run `make lib.nfv.config` with a Solarflare NIC. Result:

```
selftest: lib.nfv.config
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_nic/x
engine: start app id0_Virtio
engine: start app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_filter/x
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: start app id0_Filter_in
load: time: 0.30s  fps: 0         fpGbps: 0.000 fpb: 0   bpp: -    sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_qos/x
engine: stop app id0_Filter_in
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
load: time: 0.30s  fps: 0         fpGbps: 0.000 fpb: 0   bpp: -    sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_tunnel/x
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: start app id0_Tunnel
engine: start app id0_ND
Sending neighbor solicitation for next-hop 2::2
load: time: 0.28s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 86   sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/scale_up/y
engine: stop app id0_Tunnel
engine: stop app id0_ND
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: start app id1_Virtio
engine: start app id1_NIC
Opened SolarFlare interface p10p1 (MAC address 01:90:f5:ef:aa:67, MTU 1514)
load: time: 0.29s  fps: 6         fpGbps: 0.000 fpb: 0   bpp: 86   sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/scale_up/x
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: reconfig app id1_NIC
Opened SolarFlare interface p10p1 (MAC address 01:90:f5:ef:aa:67, MTU 1514)
engine: start app id63_Virtio
engine: start app id48_NIC
Opened SolarFlare interface p10p1 (MAC address 30:90:f5:ef:aa:67, MTU 1514)
engine: start app id10_NIC
Opened SolarFlare interface p10p1 (MAC address 0a:90:f5:ef:aa:67, MTU 1514)
engine: start app id52_Virtio
engine: start app id6_NIC
Opened SolarFlare interface p10p1 (MAC address 06:90:f5:ef:aa:67, MTU 1514)
engine: start app id45_NIC
Opened SolarFlare interface p10p1 (MAC address 2d:90:f5:ef:aa:67, MTU 1514)
engine: start app id34_NIC
Opened SolarFlare interface p10p1 (MAC address 22:90:f5:ef:aa:67, MTU 1514)
engine: start app id35_NIC
Opened SolarFlare interface p10p1 (MAC address 23:90:f5:ef:aa:67, MTU 1514)
engine: start app id40_NIC
Opened SolarFlare interface p10p1 (MAC address 28:90:f5:ef:aa:67, MTU 1514)
engine: start app id61_Virtio
engine: start app id61_NIC
Opened SolarFlare interface p10p1 (MAC address 3d:90:f5:ef:aa:67, MTU 1514)
engine: start app id60_Virtio
engine: start app id49_Virtio
engine: start app id59_Virtio
engine: start app id12_NIC
Opened SolarFlare interface p10p1 (MAC address 0c:90:f5:ef:aa:67, MTU 1514)
engine: start app id42_NIC
Opened SolarFlare interface p10p1 (MAC address 2a:90:f5:ef:aa:67, MTU 1514)
engine: start app id38_NIC
Opened SolarFlare interface p10p1 (MAC address 26:90:f5:ef:aa:67, MTU 1514)
engine: start app id24_NIC
Opened SolarFlare interface p10p1 (MAC address 18:90:f5:ef:aa:67, MTU 1514)
engine: start app id58_Virtio
engine: start app id39_Virtio
engine: start app id51_NIC
Opened SolarFlare interface p10p1 (MAC address 33:90:f5:ef:aa:67, MTU 1514)
engine: start app id57_Virtio
engine: start app id46_NIC
Opened SolarFlare interface p10p1 (MAC address 2e:90:f5:ef:aa:67, MTU 1514)
engine: start app id3_NIC
Opened SolarFlare interface p10p1 (MAC address 03:90:f5:ef:aa:67, MTU 1514)
engine: start app id56_NIC
Opened SolarFlare interface p10p1 (MAC address 38:90:f5:ef:aa:67, MTU 1514)
engine: start app id55_Virtio
engine: start app id45_Virtio
engine: start app id28_NIC
Opened SolarFlare interface p10p1 (MAC address 1c:90:f5:ef:aa:67, MTU 1514)
engine: start app id25_NIC
Opened SolarFlare interface p10p1 (MAC address 19:90:f5:ef:aa:67, MTU 1514)
engine: start app id15_Virtio
engine: start app id41_NIC
Opened SolarFlare interface p10p1 (MAC address 29:90:f5:ef:aa:67, MTU 1514)
engine: start app id35_Virtio
engine: start app id25_Virtio
engine: start app id54_Virtio
engine: start app id36_NIC
Opened SolarFlare interface p10p1 (MAC address 24:90:f5:ef:aa:67, MTU 1514)
engine: start app id42_Virtio
engine: start app id32_NIC
Opened SolarFlare interface p10p1 (MAC address 20:90:f5:ef:aa:67, MTU 1514)
engine: start app id18_NIC
Opened SolarFlare interface p10p1 (MAC address 12:90:f5:ef:aa:67, MTU 1514)
engine: start app id50_Virtio
engine: start app id62_Virtio
engine: start app id12_Virtio
engine: start app id52_NIC
Opened SolarFlare interface p10p1 (MAC address 34:90:f5:ef:aa:67, MTU 1514)
engine: start app id32_Virtio
engine: start app id22_Virtio
engine: start app id51_Virtio
engine: start app id53_NIC
Opened SolarFlare interface p10p1 (MAC address 35:90:f5:ef:aa:67, MTU 1514)
engine: start app id60_NIC
Opened SolarFlare interface p10p1 (MAC address 3c:90:f5:ef:aa:67, MTU 1514)
engine: start app id8_NIC
Opened SolarFlare interface p10p1 (MAC address 08:90:f5:ef:aa:67, MTU 1514)
engine: start app id31_NIC
Opened SolarFlare interface p10p1 (MAC address 1f:90:f5:ef:aa:67, MTU 1514)
engine: start app id33_NIC
Opened SolarFlare interface p10p1 (MAC address 21:90:f5:ef:aa:67, MTU 1514)
engine: start app id37_Virtio
engine: start app id27_Virtio
engine: start app id20_Virtio
engine: start app id30_Virtio
engine: start app id50_NIC
Opened SolarFlare interface p10p1 (MAC address 32:90:f5:ef:aa:67, MTU 1514)
engine: start app id10_Virtio
engine: start app id53_Virtio
engine: start app id47_Virtio
engine: start app id23_NIC
Opened SolarFlare interface p10p1 (MAC address 17:90:f5:ef:aa:67, MTU 1514)
engine: start app id43_NIC
Opened SolarFlare interface p10p1 (MAC address 2b:90:f5:ef:aa:67, MTU 1514)
engine: start app id13_Virtio
engine: start app id24_Virtio
engine: start app id14_Virtio
engine: start app id21_NIC
Opened SolarFlare interface p10p1 (MAC address 15:90:f5:ef:aa:67, MTU 1514)
engine: start app id7_Virtio
engine: start app id6_Virtio
engine: start app id5_Virtio
engine: start app id63_NIC
Opened SolarFlare interface p10p1 (MAC address 3f:90:f5:ef:aa:67, MTU 1514)
engine: start app id21_Virtio
engine: start app id31_Virtio
engine: start app id62_NIC
Opened SolarFlare interface p10p1 (MAC address 3e:90:f5:ef:aa:67, MTU 1514)
engine: start app id11_Virtio
engine: start app id59_NIC
Opened SolarFlare interface p10p1 (MAC address 3b:90:f5:ef:aa:67, MTU 1514)
engine: start app id13_NIC
Opened SolarFlare interface p10p1 (MAC address 0d:90:f5:ef:aa:67, MTU 1514)
engine: start app id41_Virtio
engine: start app id57_NIC
Opened SolarFlare interface p10p1 (MAC address 39:90:f5:ef:aa:67, MTU 1514)
engine: start app id44_Virtio
engine: start app id9_Virtio
engine: start app id8_Virtio
engine: start app id44_NIC
Opened SolarFlare interface p10p1 (MAC address 2c:90:f5:ef:aa:67, MTU 1514)
engine: start app id43_Virtio
engine: start app id49_NIC
Opened SolarFlare interface p10p1 (MAC address 31:90:f5:ef:aa:67, MTU 1514)
engine: start app id4_Virtio
engine: start app id4_NIC
Opened SolarFlare interface p10p1 (MAC address 04:90:f5:ef:aa:67, MTU 1514)
engine: start app id2_Virtio
engine: start app id40_Virtio
engine: start app id47_NIC
Opened SolarFlare interface p10p1 (MAC address 2f:90:f5:ef:aa:67, MTU 1514)
engine: start app id58_NIC
Opened SolarFlare interface p10p1 (MAC address 3a:90:f5:ef:aa:67, MTU 1514)
engine: start app id33_Virtio
engine: start app id23_Virtio
engine: start app id34_Virtio
engine: start app id29_NIC
Opened SolarFlare interface p10p1 (MAC address 1d:90:f5:ef:aa:67, MTU 1514)
engine: start app id37_NIC
Opened SolarFlare interface p10p1 (MAC address 25:90:f5:ef:aa:67, MTU 1514)
engine: start app id54_NIC
Opened SolarFlare interface p10p1 (MAC address 36:90:f5:ef:aa:67, MTU 1514)
engine: start app id36_Virtio
engine: start app id26_NIC
Opened SolarFlare interface p10p1 (MAC address 1a:90:f5:ef:aa:67, MTU 1514)
engine: start app id16_Virtio
engine: start app id39_NIC
Opened SolarFlare interface p10p1 (MAC address 27:90:f5:ef:aa:67, MTU 1514)
engine: start app id38_Virtio
engine: start app id46_Virtio
engine: start app id56_Virtio
engine: start app id28_Virtio
engine: start app id27_NIC
Opened SolarFlare interface p10p1 (MAC address 1b:90:f5:ef:aa:67, MTU 1514)
engine: start app id48_Virtio
engine: start app id30_NIC
Opened SolarFlare interface p10p1 (MAC address 1e:90:f5:ef:aa:67, MTU 1514)
engine: start app id16_NIC
Opened SolarFlare interface p10p1 (MAC address 10:90:f5:ef:aa:67, MTU 1514)
engine: start app id5_NIC
Opened SolarFlare interface p10p1 (MAC address 05:90:f5:ef:aa:67, MTU 1514)
engine: start app id19_NIC
Opened SolarFlare interface p10p1 (MAC address 13:90:f5:ef:aa:67, MTU 1514)
engine: start app id17_Virtio
engine: start app id22_NIC
Opened SolarFlare interface p10p1 (MAC address 16:90:f5:ef:aa:67, MTU 1514)
engine: start app id15_NIC
Opened SolarFlare interface p10p1 (MAC address 0f:90:f5:ef:aa:67, MTU 1514)
engine: start app id18_Virtio
engine: start app id20_NIC
Opened SolarFlare interface p10p1 (MAC address 14:90:f5:ef:aa:67, MTU 1514)
engine: start app id2_NIC
Opened SolarFlare interface p10p1 (MAC address 02:90:f5:ef:aa:67, MTU 1514)
engine: start app id9_NIC
Opened SolarFlare interface p10p1 (MAC address 09:90:f5:ef:aa:67, MTU 1514)
engine: start app id19_Virtio
engine: start app id29_Virtio
engine: start app id17_NIC
Opened SolarFlare interface p10p1 (MAC address 11:90:f5:ef:aa:67, MTU 1514)
engine: start app id11_NIC
Opened SolarFlare interface p10p1 (MAC address 0b:90:f5:ef:aa:67, MTU 1514)
engine: start app id55_NIC
Opened SolarFlare interface p10p1 (MAC address 37:90:f5:ef:aa:67, MTU 1514)
engine: start app id7_NIC
Opened SolarFlare interface p10p1 (MAC address 07:90:f5:ef:aa:67, MTU 1514)
engine: start app id14_NIC
Segmentation fault (core dumped)
EXITCODE: 139
```

**Comment by dpino** [comment]
I'm not sure about how to run it with _make_. I run it as:

```bash
$ sudo SNABB_TEST_INTEL10G_PCIDEVA=0000:86:00.0 ./snabb snsh -t program.snabbnfv.nfvconfig
```

This is what I got. Apparently it seems to work fine. Machine is grindelwald.

```
selftest: lib.nfv.config
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_nic/x
engine: start app id0_Virtio
engine: start app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_filter/x
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: start app id0_Filter_in
load: time: 0.30s  fps: 0         fpGbps: 0.000 fpb: 0   bpp: -    sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_qos/x
engine: stop app id0_Filter_in
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
load: time: 0.27s  fps: 0         fpGbps: 0.000 fpb: 0   bpp: -    sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/switch_tunnel/x
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: start app id0_Tunnel
engine: start app id0_ND
Sending neighbor solicitation for next-hop 2::2
load: time: 0.29s  fps: 3         fpGbps: 0.000 fpb: 0   bpp: 86   sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/scale_up/y
engine: stop app id0_Tunnel
engine: stop app id0_ND
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: start app id1_Virtio
engine: start app id1_NIC
Opened SolarFlare interface p10p1 (MAC address 01:90:f5:ef:aa:67, MTU 1514)
load: time: 0.29s  fps: 7         fpGbps: 0.000 fpb: 0   bpp: 86   sleep: 100 us
testing:	program/snabbnfv/test_fixtures/nfvconfig/scale_up/x
engine: reconfig app id0_NIC
Opened SolarFlare interface p10p1 (MAC address 00:90:f5:ef:aa:67, MTU 1514)
engine: reconfig app id1_NIC
Opened SolarFlare interface p10p1 (MAC address 01:90:f5:ef:aa:67, MTU 1514)
engine: start app id63_Virtio
engine: start app id48_NIC
Opened SolarFlare interface p10p1 (MAC address 30:90:f5:ef:aa:67, MTU 1514)
engine: start app id10_NIC
Opened SolarFlare interface p10p1 (MAC address 0a:90:f5:ef:aa:67, MTU 1514)
engine: start app id52_Virtio
engine: start app id6_NIC
Opened SolarFlare interface p10p1 (MAC address 06:90:f5:ef:aa:67, MTU 1514)
engine: start app id45_NIC
Opened SolarFlare interface p10p1 (MAC address 2d:90:f5:ef:aa:67, MTU 1514)
engine: start app id34_NIC
Opened SolarFlare interface p10p1 (MAC address 22:90:f5:ef:aa:67, MTU 1514)
engine: start app id35_NIC
Opened SolarFlare interface p10p1 (MAC address 23:90:f5:ef:aa:67, MTU 1514)
engine: start app id40_NIC
Opened SolarFlare interface p10p1 (MAC address 28:90:f5:ef:aa:67, MTU 1514)
engine: start app id61_Virtio
engine: start app id61_NIC
Opened SolarFlare interface p10p1 (MAC address 3d:90:f5:ef:aa:67, MTU 1514)
engine: start app id60_Virtio
engine: start app id49_Virtio
engine: start app id59_Virtio
engine: start app id12_NIC
Opened SolarFlare interface p10p1 (MAC address 0c:90:f5:ef:aa:67, MTU 1514)
engine: start app id42_NIC
Opened SolarFlare interface p10p1 (MAC address 2a:90:f5:ef:aa:67, MTU 1514)
engine: start app id38_NIC
Opened SolarFlare interface p10p1 (MAC address 26:90:f5:ef:aa:67, MTU 1514)
engine: start app id24_NIC
Opened SolarFlare interface p10p1 (MAC address 18:90:f5:ef:aa:67, MTU 1514)
engine: start app id58_Virtio
engine: start app id39_Virtio
engine: start app id51_NIC
Opened SolarFlare interface p10p1 (MAC address 33:90:f5:ef:aa:67, MTU 1514)
engine: start app id57_Virtio
engine: start app id46_NIC
Opened SolarFlare interface p10p1 (MAC address 2e:90:f5:ef:aa:67, MTU 1514)
engine: start app id3_NIC
Opened SolarFlare interface p10p1 (MAC address 03:90:f5:ef:aa:67, MTU 1514)
engine: start app id56_NIC
Opened SolarFlare interface p10p1 (MAC address 38:90:f5:ef:aa:67, MTU 1514)
engine: start app id55_Virtio
engine: start app id45_Virtio
engine: start app id28_NIC
Opened SolarFlare interface p10p1 (MAC address 1c:90:f5:ef:aa:67, MTU 1514)
engine: start app id25_NIC
Opened SolarFlare interface p10p1 (MAC address 19:90:f5:ef:aa:67, MTU 1514)
engine: start app id15_Virtio
engine: start app id41_NIC
Opened SolarFlare interface p10p1 (MAC address 29:90:f5:ef:aa:67, MTU 1514)
engine: start app id35_Virtio
engine: start app id25_Virtio
engine: start app id54_Virtio
engine: start app id36_NIC
Opened SolarFlare interface p10p1 (MAC address 24:90:f5:ef:aa:67, MTU 1514)
engine: start app id42_Virtio
engine: start app id32_NIC
Opened SolarFlare interface p10p1 (MAC address 20:90:f5:ef:aa:67, MTU 1514)
engine: start app id18_NIC
Opened SolarFlare interface p10p1 (MAC address 12:90:f5:ef:aa:67, MTU 1514)
engine: start app id50_Virtio
engine: start app id62_Virtio
engine: start app id12_Virtio
engine: start app id52_NIC
Opened SolarFlare interface p10p1 (MAC address 34:90:f5:ef:aa:67, MTU 1514)
engine: start app id32_Virtio
engine: start app id22_Virtio
engine: start app id51_Virtio
engine: start app id53_NIC
Opened SolarFlare interface p10p1 (MAC address 35:90:f5:ef:aa:67, MTU 1514)
engine: start app id60_NIC
Opened SolarFlare interface p10p1 (MAC address 3c:90:f5:ef:aa:67, MTU 1514)
engine: start app id8_NIC
Opened SolarFlare interface p10p1 (MAC address 08:90:f5:ef:aa:67, MTU 1514)
engine: start app id31_NIC
Opened SolarFlare interface p10p1 (MAC address 1f:90:f5:ef:aa:67, MTU 1514)
engine: start app id33_NIC
Opened SolarFlare interface p10p1 (MAC address 21:90:f5:ef:aa:67, MTU 1514)
engine: start app id37_Virtio
engine: start app id27_Virtio
engine: start app id20_Virtio
engine: start app id30_Virtio
engine: start app id50_NIC
Opened SolarFlare interface p10p1 (MAC address 32:90:f5:ef:aa:67, MTU 1514)
engine: start app id10_Virtio
engine: start app id53_Virtio
engine: start app id47_Virtio
engine: start app id23_NIC
Opened SolarFlare interface p10p1 (MAC address 17:90:f5:ef:aa:67, MTU 1514)
engine: start app id43_NIC
Opened SolarFlare interface p10p1 (MAC address 2b:90:f5:ef:aa:67, MTU 1514)
engine: start app id13_Virtio
engine: start app id24_Virtio
engine: start app id14_Virtio
engine: start app id21_NIC
Opened SolarFlare interface p10p1 (MAC address 15:90:f5:ef:aa:67, MTU 1514)
engine: start app id7_Virtio
engine: start app id6_Virtio
engine: start app id5_Virtio
engine: start app id63_NIC
Opened SolarFlare interface p10p1 (MAC address 3f:90:f5:ef:aa:67, MTU 1514)
engine: start app id21_Virtio
engine: start app id31_Virtio
engine: start app id62_NIC
Opened SolarFlare interface p10p1 (MAC address 3e:90:f5:ef:aa:67, MTU 1514)
engine: start app id11_Virtio
engine: start app id59_NIC
Opened SolarFlare interface p10p1 (MAC address 3b:90:f5:ef:aa:67, MTU 1514)
engine: start app id13_NIC
Opened SolarFlare interface p10p1 (MAC address 0d:90:f5:ef:aa:67, MTU 1514)
engine: start app id41_Virtio
engine: start app id57_NIC
Opened SolarFlare interface p10p1 (MAC address 39:90:f5:ef:aa:67, MTU 1514)
engine: start app id44_Virtio
engine: start app id9_Virtio
engine: start app id8_Virtio
engine: start app id44_NIC
Opened SolarFlare interface p10p1 (MAC address 2c:90:f5:ef:aa:67, MTU 1514)
engine: start app id43_Virtio
engine: start app id49_NIC
Opened SolarFlare interface p10p1 (MAC address 31:90:f5:ef:aa:67, MTU 1514)
engine: start app id4_Virtio
engine: start app id4_NIC
Opened SolarFlare interface p10p1 (MAC address 04:90:f5:ef:aa:67, MTU 1514)
engine: start app id2_Virtio
engine: start app id40_Virtio
engine: start app id47_NIC
Opened SolarFlare interface p10p1 (MAC address 2f:90:f5:ef:aa:67, MTU 1514)
engine: start app id58_NIC
Opened SolarFlare interface p10p1 (MAC address 3a:90:f5:ef:aa:67, MTU 1514)
engine: start app id33_Virtio
engine: start app id23_Virtio
engine: start app id34_Virtio
engine: start app id29_NIC
Opened SolarFlare interface p10p1 (MAC address 1d:90:f5:ef:aa:67, MTU 1514)
engine: start app id37_NIC
Opened SolarFlare interface p10p1 (MAC address 25:90:f5:ef:aa:67, MTU 1514)
engine: start app id54_NIC
Opened SolarFlare interface p10p1 (MAC address 36:90:f5:ef:aa:67, MTU 1514)
engine: start app id36_Virtio
engine: start app id26_NIC
Opened SolarFlare interface p10p1 (MAC address 1a:90:f5:ef:aa:67, MTU 1514)
engine: start app id16_Virtio
engine: start app id39_NIC
Opened SolarFlare interface p10p1 (MAC address 27:90:f5:ef:aa:67, MTU 1514)
engine: start app id38_Virtio
engine: start app id46_Virtio
engine: start app id56_Virtio
engine: start app id28_Virtio
engine: start app id27_NIC
Opened SolarFlare interface p10p1 (MAC address 1b:90:f5:ef:aa:67, MTU 1514)
engine: start app id48_Virtio
engine: start app id30_NIC
Opened SolarFlare interface p10p1 (MAC address 1e:90:f5:ef:aa:67, MTU 1514)
engine: start app id16_NIC
Opened SolarFlare interface p10p1 (MAC address 10:90:f5:ef:aa:67, MTU 1514)
engine: start app id5_NIC
Opened SolarFlare interface p10p1 (MAC address 05:90:f5:ef:aa:67, MTU 1514)
engine: start app id19_NIC
Opened SolarFlare interface p10p1 (MAC address 13:90:f5:ef:aa:67, MTU 1514)
engine: start app id17_Virtio
engine: start app id22_NIC
Opened SolarFlare interface p10p1 (MAC address 16:90:f5:ef:aa:67, MTU 1514)
engine: start app id15_NIC
Opened SolarFlare interface p10p1 (MAC address 0f:90:f5:ef:aa:67, MTU 1514)
engine: start app id18_Virtio
engine: start app id20_NIC
Opened SolarFlare interface p10p1 (MAC address 14:90:f5:ef:aa:67, MTU 1514)
engine: start app id2_NIC
Opened SolarFlare interface p10p1 (MAC address 02:90:f5:ef:aa:67, MTU 1514)
engine: start app id9_NIC
Opened SolarFlare interface p10p1 (MAC address 09:90:f5:ef:aa:67, MTU 1514)
engine: start app id19_Virtio
engine: start app id29_Virtio
engine: start app id17_NIC
Opened SolarFlare interface p10p1 (MAC address 11:90:f5:ef:aa:67, MTU 1514)
engine: start app id11_NIC
Opened SolarFlare interface p10p1 (MAC address 0b:90:f5:ef:aa:67, MTU 1514)
engine: start app id55_NIC
Opened SolarFlare interface p10p1 (MAC address 37:90:f5:ef:aa:67, MTU 1514)
engine: start app id7_NIC
Opened SolarFlare interface p10p1 (MAC address 07:90:f5:ef:aa:67, MTU 1514)
engine: start app id14_NIC
Opened SolarFlare interface p10p1 (MAC address 0e:90:f5:ef:aa:67, MTU 1514)
engine: start app id3_Virtio
engine: start app id26_Virtio
Sending neighbor solicitation for next-hop 2::2
```

**Comment by eugeneia** [comment]
@dpino Interesting! If I run the selftest using make like so

```
sudo SNABB_TEST_INTEL10G_PCIDEVA=0000:86:00.0 make program.snabbnfv.nfvconfig
```

it will produce the segfault as described. That's definitely a hint to follow.

**Comment by dpino** [comment]
Apparently the other command works, but actually status code is 139 (SIGSEGV):

```bash
Opened SolarFlare interface p10p1 (MAC address 0e:90:f5:ef:aa:67, MTU 1514)
engine: start app id3_Virtio
engine: start app id26_Virtio
Sending neighbor solicitation for next-hop 2::2
dpino@grindelwald src (master) $ echo $?
139
```

**Comment by eugeneia** [comment]
Fixed by #664
